app/grandchallenge/components/views.py

```diff
@@ -337,3 +342,75 @@ def get_queryset(self):
         return queryset.prefetch_related(
             "values", "values__image", "values__interface"
         )
+
+
+class CIVSetBulkDeleteView(LoginRequiredMixin, FormView):
+    form_class = CIVSetDeleteForm
+    model = None
+    template_name = "components/civ_set_delete_confirm.html"
+
+    @property
+    def base_object(self):
+        raise NotImplementedError
+
+    def get_queryset(self, *args, **kwargs):
+        # subset by permission
+        permission = f"{self.model._meta.app_label}.delete_{self.model._meta.model_name}"
+        return get_objects_for_user(
+            user=self.request.user,
+            perms=[permission],
+            klass=self.base_object.civ_sets_related_manager.all(),
+        )
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context.update(
+            {
+                "base_object": self.base_object,
+                "delete_count": (
+                    self.selected_objects.count()
+                    if self.selected_objects
+                    else 0
+                ),
+            }
+        )
+        return context
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs.update(
+            {
+                "queryset": (
+                    self.selected_objects
+                    if self.selected_objects
+                    else self.get_queryset()
+                ),
+            }
+        )
+        return kwargs
+
+    @cached_property
+    def selected_objects(self):

```

Comment:
Actually, is there a reason for having this as a separate property rather than doing in `get_queryset`? If it were done there you could avoid all of the `self.selected_objects if self.selected_objects else self.get_queryset()`.

Reply:
I found it clearer to seperate, but it can be combined. We just need to make sure not to return an empty queryset on POST then because else validation on the form will fail. So then technically "delete_count" is no longer a delete count on POST.  That doesn't really matter since we only use that in the context of a GET request, but I have a slight preference for leaving the methods separate.